spec/ffi/callback_spec.rb

```diff
@@ -771,3 +771,96 @@ module LibTestStdcall
     end
   end
 end
+
+describe "Callback interop" do
+  require 'fiddle'
+  require 'fiddle/import'
+  require 'timeout'
+
+  module LibTestFFI
+    extend FFI::Library
+    ffi_lib TestLibrary::PATH
+    attach_function :testCallbackVrV, :testClosureVrV, [ :pointer ], :void
+    attach_function :testCallbackVrV_blocking, :testClosureVrV, [ :pointer ], :void, blocking: true
+  end
+
+  module LibTestFiddle
+    extend Fiddle::Importer
+    dlload TestLibrary::PATH
+    extern 'void testClosureVrV(void *fp)'
+  end
+
+  def assert_callback_in_same_thread_called_once
+    called = 0
+    thread = nil
+    yield proc {
+      called += 1
+      thread = Thread.current
+    }
+    expect(called).to eq(1)
+    expect(thread).to eq(Thread.current)
+  end
+
+  it "from ffi to ffi" do
+    assert_callback_in_same_thread_called_once do |block|
+      func = FFI::Function.new(:void, [:pointer], &block)
+      LibTestFFI.testCallbackVrV(FFI::Pointer.new(func.to_i))
+    end
+  end
+
+  it "from ffi to ffi with blocking:true" do
+    assert_callback_in_same_thread_called_once do |block|
+      func = FFI::Function.new(:void, [:pointer], &block)
+      LibTestFFI.testCallbackVrV_blocking(FFI::Pointer.new(func.to_i))
+    end
+  end
+
+  # https://github.com/ffi/ffi/issues/527
+  if RUBY_VERSION.split('.').map(&:to_i).pack("C*") >= [2,3,0].pack("C*") || RUBY_PLATFORM =~ /java/
+    it "from fiddle to ffi" do
+      assert_callback_in_same_thread_called_once do |block|
+        func = FFI::Function.new(:void, [:pointer], &block)
+        LibTestFiddle.testClosureVrV(Fiddle::Pointer[func.to_i])
+      end
+    end
+  end
+
+  it "from ffi to fiddle" do
+    assert_callback_in_same_thread_called_once do |block|
+      func = LibTestFiddle.bind_function(:cbVrV, Fiddle::TYPE_VOID, [], &block)
+      LibTestFFI.testCallbackVrV(FFI::Pointer.new(func.to_i))
+    end
+  end
+
+  it "from ffi to fiddle with blocking:true" do
+    assert_callback_in_same_thread_called_once do |block|
+      func = LibTestFiddle.bind_function(:cbVrV, Fiddle::TYPE_VOID, [], &block)
+      LibTestFFI.testCallbackVrV_blocking(FFI::Pointer.new(func.to_i))
+    end
+  end
+
+  it "from fiddle to fiddle" do
+    assert_callback_in_same_thread_called_once do |block|
+      func = LibTestFiddle.bind_function(:cbVrV, Fiddle::TYPE_VOID, [], &block)
+      LibTestFiddle.testClosureVrV(Fiddle::Pointer[func.to_i])
+    end
+  end
+
+  # https://github.com/ffi/ffi/issues/527
+  if RUBY_ENGINE == 'ruby' && RUBY_VERSION.split('.').map(&:to_i).pack("C*") >= [2,3,0].pack("C*")
```

Comment:
Since we have seen ruby-2.1.10 and I know you as a very exact character, I thought it's best to compare exactly...

rbx doesn't support fiddle, so that these tests fail. The embed_test seems to work in rbx after some tweeking. May I adjust the tests for compatibility with rbx?